code/modules/projectiles/guns/projectile/revolver.dm

```diff
@@ -147,37 +166,44 @@
 
-	if(!getAmmo())
+	var/obj/item/ammo_casing/AC = loaded[1]
+	if(isliving(target) && isliving(user) && target == user)
+		var/datum/organ/external/affecting = user.zone_sel.selecting
+		if(affecting == "head" || "mouth")
+			if(!AC || !AC.BB)
+				user.visible_message("<span class='warning'>*click*</span>", "<span class='warning'>*click*</span>")
+				playsound(user, 'sound/weapons/empty.ogg', 100, 1)
+				loaded.Cut(1,2)
+				loaded += AC
+				return
+			if(AC.BB)
+				in_chamber = AC.BB //Load projectile into chamber.
+				AC.BB.loc = src //Set projectile loc to gun.
+				AC.BB = null //Empty casings
+				AC.loc = src.loc //remove empty casing
+				AC.update_icon()
+			if(!in_chamber)
+				return
+			var/obj/item/projectile/P = new AC.projectile_type
+			playsound(user, fire_sound, 50, 1)
+			user.visible_message("<span class='danger'>[user.name] fires [src] [affecting == "head" ? "at their head" : "in their mouth"]!</span>", "<span class='danger'>You fire [src] [affecting == "head" ? "at your head" : "in your mouth"]!</span>", "You hear a [istype(in_chamber, /obj/item/projectile/beam) ? "laser blast" : "gunshot"]!")
+			if(!P.nodamage)
+				affecting = "head"
+				user.apply_damage(300, BRUTE, affecting) // You are dead, dead, dead.
+			in_chamber = null
+			loaded.Cut(1,2)
+			loaded += null //replace fired round
+
+	..()
+
+/obj/item/weapon/gun/projectile/russian/Fire(atom/target as mob|obj|turf|area, mob/living/user as mob|obj, params, reflex = 0, struggle = 0)
+	var/obj/item/ammo_casing/AC = loaded[1]
+	if(!AC || !AC.BB)
 		user.visible_message("<span class='warning'>*click*</span>", "<span class='warning'>*click*</span>")
 		playsound(user, 'sound/weapons/empty.ogg', 100, 1)
+		loaded.Cut(1,2)
+		loaded += AC
 		return
 
-	if(isliving(target) && isliving(user))
-		if(target == user)
-			var/datum/organ/external/affecting = user.zone_sel.selecting
-			if(affecting == "head")
-
-				var/obj/item/ammo_casing/AC = loaded[1]
-				if(!AC || !AC.BB)
-					user.visible_message("<span class='warning'>*click*</span>", "<span class='warning'>*click*</span>")
-					playsound(user, 'sound/weapons/empty.ogg', 100, 1)
-					loaded.Cut(1,2)
-					loaded += AC
-					return
-				if(AC.BB)
-					in_chamber = AC.BB //Load projectile into chamber.
-					AC.BB.loc = src //Set projectile loc to gun.
-					AC.BB = null //Empty casings
-					AC.update_icon()
-				if(!in_chamber)
-					return
-				var/obj/item/projectile/P = new AC.projectile_type
-				playsound(user, fire_sound, 50, 1)
-				user.visible_message("<span class='danger'>[user.name] fires [src] at \his head!</span>", "<span class='danger'>You fire [src] at your head!</span>", "You hear a [istype(in_chamber, /obj/item/projectile/beam) ? "laser blast" : "gunshot"]!")
-				if(!P.nodamage)
-					user.apply_damage(300, BRUTE, affecting) // You are dead, dead, dead.
-				in_chamber = null
-				loaded.Cut(1,2)
-				loaded += AC
-				return
 	..()
+	loaded += null //replace fired round
```

Comment:
Here's a potential problem.
If loaded contains a bunch of null entries, and getAmmo() just returns the length of the loaded list, and loaded is full of 6 null entries, you'd end up being unable to load anything because it'd be full of nothing.

Just a POTENTIAL problem. Also curious as to what happens when you empty the revolver. Does it dump null shell casings on the ground?